src/frontend/src/optimizer/plan_node/logical_limit.rs

```diff
@@ -33,17 +35,25 @@ use crate::utils::{ColIndexMapping, Condition};
 pub struct LogicalLimit {
     pub base: PlanBase<Logical>,
     pub(super) core: generic::Limit<PlanRef>,
+
+    /// Whether to report an error when the number of rows from the input
+    /// exceeds the limit.
+    pub check_exceeding: bool,
```

Comment:
I feel this can lead to confusion: When I see this, I feel: why would we ever want to check exceeding? There seems to be no general use case, but only "max one row" is needed?

Reply:
Yeah, you're pretty right. 😕